src/components/validation/validators/component_spec/mod.rs

```diff
@@ -263,7 +283,15 @@ fn compare_actual_to_expected(
 
     info!("{}: expected {}, actual {}.", metric_type, expected, actual,);
 
-    if actual != expected {
+    if actual != expected &&
+        // This is a bit messy. The issue is that EstimatedJsonSizeOf can be called by a component
+        // on an event array, or on a single event. And we have no way of knowing which that is.
+        // By default the input driver for the framework is not assuming it is an array, so we
+        // check here if it matches what the array scenario would be, which is to add the size of
+        // the brackets, for each event.
+        (metric_type != &ComponentMetricType::EventsReceivedBytes
```

Comment:
Is there test coverage for this case?

Reply:
This is a test framework code, so this is the test coverage...

This branch contains changes to the framework that are utilized in downstream branches. it was done this way to separate the framework changes from usage to make it more reviewable.

Reply:
Ha my bad, thanks for clarifying. Pretty sure I was looking at a component validation PR just a few days ago, but had forgotten that context